tests/Workflow/MoveJatsAssetsTest.php

```diff
@@ -369,4 +369,196 @@ public function it_fails_if_an_invalid_content_type_is_returned() : void
 
         $mover->onManipulate($event);
     }
+
+    /**
+     * @test
+     */
+    public function it_uses_the_extension_if_no_content_type_is_returned() : void
+    {
+        $mover = new MoveJatsAssets('.+', 'http://public-assets/path', $this->filesystem, $this->client);
+
+        $document = FluentDOM::load(
+            <<<XML
+<item xmlns="http://libero.pub">
+    <article xmlns="http://jats.nlm.nih.gov" xmlns:xlink="http://www.w3.org/1999/xlink">
+        <body>
+            <graphic xlink:href="http://origin-assets/assets/figure.jpg"/>
+        </body>
+    </article>
+</item>
+XML
+        );
+
+        $marking = new Marking();
+        $task = new PutTask('service', ItemId::fromString('id'), ItemVersionNumber::fromInt(1), $document);
+        $transition = new Transition('transition', 'place1', 'place2');
+
+        $event = new Event($task, $marking, $transition);
+
+        $this->mock->save(
+            new Request('GET', 'http://origin-assets/assets/figure.jpg'),
+            new Response(200, [], 'figure')
+        );
+
+        $mover->onManipulate($event);
+
+        $expected = FluentDOM::load(
+            <<<XML
+<item xmlns="http://libero.pub">
+    <article xmlns="http://jats.nlm.nih.gov" xmlns:xlink="http://www.w3.org/1999/xlink">
+        <body>
+            <graphic xlink:href="http://public-assets/path/id/v1/cb071d80d1a54f21c8867a038f6a6c66.jpeg"
+                mimetype="image" mime-subtype="jpeg"/>
+        </body>
+    </article>
+</item>
+XML
+        );
+
+        $this->assertXmlStringEqualsXmlString($expected, $task->getDocument());
+    }
+
+    /**
+     * @test
+     */
+    public function it_uses_the_extension_if_an_invalid_content_type_is_returned() : void
+    {
+        $mover = new MoveJatsAssets('.+', 'http://public-assets/path', $this->filesystem, $this->client);
+
+        $document = FluentDOM::load(
+            <<<XML
+<item xmlns="http://libero.pub">
+    <article xmlns="http://jats.nlm.nih.gov" xmlns:xlink="http://www.w3.org/1999/xlink">
+        <body>
+            <graphic xlink:href="http://origin-assets/assets/figure.jpg"/>
+        </body>
+    </article>
+</item>
+XML
+        );
+
+        $marking = new Marking();
```

Comment:
there is some shared setup that makes the tests long, obscuring their main point of turning headers or file extensions into the right `xlink:href` attribute. What can be extracted into private methods?